src/astarte-charts/providers/deviceDatastreamObject.ts

```diff
@@ -0,0 +1,76 @@
+/*
+   This file is part of Astarte.
+
+   Copyright 2020 Ispirata Srl
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import AstarteClient from 'astarte-client';
+import type { AstarteDatastreamObjectData } from 'astarte-client';
+
+import {
+  AstarteChartProvider,
+  AstarteChartValueObjectTimestamped,
+  AstarteChartListValue,
+} from '../index';
+
+type AstarteDeviceDatastreamObjectProviderValue = AstarteChartListValue<
+  AstarteChartValueObjectTimestamped
+>;
+
+interface AstarteDeviceDatastreamObjectProviderParams {
+  name?: string;
+  deviceId: string;
+  interfaceName: string;
+  endpoint: string;
+}
+
+export class AstarteDeviceDatastreamObjectProvider
+  implements AstarteChartProvider<AstarteDeviceDatastreamObjectProviderValue> {
```

Comment:
those sure are super long names. Let's find something shorter or use aliases for things we're not going to export

Reply:
Readability is a good point to strive for, I must agree. I chose to go with `ProviderParams` and `ProviderValue`, instead of `AstarteDeviceDatastreamObjectProviderParams` and `AstarteDeviceDatastreamObjectProviderValue`. Same for the other providers.